session-recordings/test/ingestion.test.ts

```diff
@@ -0,0 +1,331 @@
+                        eventId: eventUuid,
+                        sessionId: sessionId,
+                        windowId: windowId,
+                        chunkIndex: '0',
+                        chunkCount: '1',
+                        teamId: '1',
+                        eventSource: '1',
+                        eventType: '3',
+                    },
+                    value: `{"uuid": "${eventUuid}", "type": 3, "data": {"source": 1, "positions": [{"x": 829, "y": 154, "id": 367, "timeOffset": 0}]}, "timestamp": 1657682896740}`,
+                },
+            ],
+        })
+
+        const sessionRecording = await waitForSessionRecording(teamId, sessionId, eventUuid)
+
+        expect(sessionRecording.events.length).toBe(1)
+    })
+
+    it('handles chunked events', async ({ producer }) => {
+        const teamId = '1'
+        const sessionId = uuidv4()
+        const windowId = uuidv4()
+        const eventUuid = uuidv4()
+
+        const fullEventString = `{"uuid": "${eventUuid}", "type": 3, "data": {"source": 1, "positions": [{"x": 829, "y": 154, "id": 367, "timeOffset": 0}]}, "timestamp": 1657682896740}`
+
+        const [first] = await producer.send({
+            topic: RECORDING_EVENTS_TOPIC,
+            messages: [
+                {
+                    partition: 0,
+                    headers: {
+                        unixTimestamp: '1657682896740',
+                        eventId: eventUuid,
+                        sessionId: sessionId,
+                        windowId: windowId,
+                        chunkIndex: '0',
+                        chunkCount: '2',
+                        teamId: '1',
+                        eventSource: '1',
+                        eventType: '3',
+                    },
+                    value: fullEventString.slice(0, 100),
+                },
+            ],
+        })
+
+        await producer.send({
+            topic: RECORDING_EVENTS_TOPIC,
+            messages: [
+                {
+                    partition: 0,
+                    headers: {
+                        unixTimestamp: '1657682896740',
+                        eventId: eventUuid,
+                        sessionId: sessionId,
+                        windowId: windowId,
+                        chunkIndex: '1',
+                        chunkCount: '2',
+                        chunkOffset: first.baseOffset,
+                        teamId: '1',
+                        eventSource: '1',
+                        eventType: '3',
+                    },
+                    value: fullEventString.slice(100),
+                },
+            ],
+        })
+
+        const sessionRecording = await waitForSessionRecording(teamId, sessionId, eventUuid)
+
+        expect(sessionRecording.events.length).toBe(1)
+        expect(sessionRecording.events[0]).toMatchObject(JSON.parse(fullEventString))
+    })
+
+    // TODO: Handle this case and re-enable the test
```

Comment:
@rcmarron what do you think the cases here are? I know we get some guarantees on this from Kafka, but I guess there might be other parts of the system that we need to consider?

Reply:
For details on guarantees perhaps https://kafka.apache.org/documentation/#semantics is useful